src/datarobotx/idp/common/hashing.py

```diff
@@ -52,6 +54,10 @@ def get_hash(*args: Any, **kwargs: Any) -> str:
         elif arg is None:
             arg = int_to_bytes(_NONE_REPRESENTATION)
         elif isinstance(arg, str):
+            for t in ALLOWED_FILE_TYPES:
+                if t in arg:
```

Comment:
Not a huge change but you could remove a for loop here by setting arg to be a pathlib.Path object. 
```python
if Path(arg).exists: 
   suffix = "".join(Path(arg).suffixes)
if suffix in ALLOWED_FILE_TYPES:
   arg = get_hash(Path(arg))
```
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffixes